src/fio/fioPlugin.js

```diff
@@ -0,0 +1,217 @@
+import { FIOSDK } from '@dapix/react-native-fio'
+import { bns } from 'biggystring'
+/**
+ * Created by paul on 8/8/17.
+ */
+// @flow
+import {
+  type EdgeCorePluginOptions,
+  type EdgeCurrencyEngine,
+  type EdgeCurrencyEngineOptions,
+  type EdgeCurrencyPlugin,
+  type EdgeEncodeUri,
+  type EdgeIo,
+  type EdgeParsedUri,
+  type EdgeWalletInfo
+} from 'edge-core-js/types'
+
+import { CurrencyPlugin } from '../common/plugin.js'
+import { getDenomInfo, getEdgeInfoServer } from '../common/utils.js'
+import { getFetchJson } from '../react-native-io.js'
+import { FioEngine } from './fioEngine'
+import { currencyInfo } from './fioInfo.js'
+
+export function checkAddress (address: string): boolean {
+  const start = address.startsWith('FIO')
+  const lenght = address.length === 53
+  return start && lenght
+}
+
+export class FioPlugin extends CurrencyPlugin {
+  otherMethods: Object
+  eosServer: Object
+
+  constructor (io: EdgeIo) {
+    super(io, 'fio', currencyInfo)
+  }
+
+  async createPrivateKey (walletType: string): Promise<Object> {
+    const type = walletType.replace('wallet:', '')
+    if (type === 'fio') {
+      const buffer = this.io.random(32)
+      return FIOSDK.createPrivateKey(buffer)
```

Comment:
I see that the format is:

```typescript
{
  fioKey: string // base58?,
  mnemonic: string
}
```

If the `fioKey` is missing, will the FIO SDK still work correctly?

Reply:
For this, I don't know the implications of this. Or how the fioKey may be missing in this scenario.

FIOSDK.createPrivateKey(buffer) does take a buffer to generate the private key.  And the FIO SDK as is, is just calling static methods.